File: test/data/languages/php/Inheritance.php
<?php
/**
 * PHP Inheritance test file for native extraction validation
 */

// Simple interface
interface Runnable {
    public function run(): void;
}

// Interface extending another
interface Runner extends Runnable {
    public function sprint(): void;
}

// Interface extending multiple interfaces
interface Athlete extends Runner, Serializable {
    public function compete(): void;
}

// Abstract base class
abstract class Animal {
    protected string $name;

    abstract public function makeSound(): string;
}

// Class extending another class
class Dog extends Animal {
    public function makeSound(): string {
        return "Woof!";
    }
}

// Class extending and implementing
class Pet extends Dog implements Serializable {
    public function serialize(): string {
        return serialize($this->name);
    }

    public function unserialize(string $data): void {
        $this->name = unserialize($data);
    }
}

// Class with multiple implements
class ServiceAnimal extends Dog implements Runnable, Serializable {
    public function run(): void {
        echo "Running...";
    }

    public function serialize(): string {
        return serialize($this->name);
    }

    public function unserialize(string $data): void {
        $this->name = unserialize($data);
    }
}

// Final class
final class Bulldog extends Dog {
    public function makeSound(): string {
        return "Gruff!";
    }
}

// Trait (PHP-specific)
trait Loggable {
    public function log(string $message): void {
        echo "[LOG] " . $message;
    }
}

// Enum implementing interface (PHP 8.1+)
enum DogBreed: string implements Serializable {
    case LABRADOR = 'labrador';
    case BULLDOG = 'bulldog';
    case POODLE = 'poodle';

    public function serialize(): string {
        return $this->value;
    }

    public function unserialize(string $data): void {
        // Enums are immutable
    }
}

File: test/data/languages/swift/Inheritance.swift
/// Swift Inheritance test file for native extraction validation

// Protocol (Swift's interface)
protocol Runnable {
    func run()
}

// Protocol extending another
protocol Runner: Runnable {
    func sprint()
}

// Protocol with multiple inheritance
protocol Athlete: Runner, CustomStringConvertible {
    func compete()
}

// Base class
class Animal {
    var name: String

    init(name: String) {
        self.name = name
    }

    func makeSound() -> String {
        return ""
    }
}

// Class extending another class
class Dog: Animal {
    override func makeSound() -> String {
        return "Woof!"
    }
}

// Class with extends and protocol conformance
class Pet: Dog, Runnable {
    func run() {
        print("\(name) is running...")
    }
}

// Class with multiple protocol conformance
class ServiceAnimal: Dog, Runnable, CustomStringConvertible {
    func run() {
        print("\(name) is running for service!")
    }

    var description: String {
        return "ServiceAnimal: \(name)"
    }
}

// Final class
final class Bulldog: Dog {
    override func makeSound() -> String {
        return "Gruff!"
    }
}

// Struct with protocol conformance
struct Point: Equatable, Hashable {
    var x: Double
    var y: Double
}

// Enum with protocol conformance
enum DogBreed: String, CaseIterable {
    case labrador = "Labrador"
    case bulldog = "Bulldog"
    case poodle = "Poodle"
}

// Actor (Swift concurrency)
actor DataManager: CustomStringConvertible {
    var data: [String] = []

    var description: String {
        return "DataManager with \(data.count) items"
    }
}

// Generic class with constraints
class Container<T: Equatable>: CustomStringConvertible {
    var items: [T] = []

    var description: String {
        return "Container with \(items.count) items"
    }
}
